homeassistant/components/fortios/device_tracker.py

```diff
@@ -0,0 +1,111 @@
+"""
+Support to use FortiOS device like FortiGate as device tracker.
+
+This component is part of the device_tracker platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/fortiosapi/
+"""
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.const import CONF_VERIFY_SSL
+
+REQUIREMENTS = ['fortiosapi==0.10.8']
+
+_LOGGER = logging.getLogger(__name__)
+DEFAULT_VERIFY_SSL = False
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean
+})
+
+
+def get_scanner(hass, config):
+    """Validate the configuration and return a FortiOSDeviceScanner."""
+    from fortiosapi import FortiOSAPI
+
+    host = config[DOMAIN].get(CONF_HOST)
+    verify_ssl = config[DOMAIN].get(CONF_VERIFY_SSL)
+    token = config[DOMAIN].get(CONF_TOKEN)
+
+    _LOGGER.debug("host : %s", host)
+    _LOGGER.debug("verify_ssl : %s", verify_ssl)
+    _LOGGER.debug("token : %s", token)
+
+    _LOGGER.debug('fortios, get_scanner')
+
+    fgt = FortiOSAPI()
+
+    try:
+        fgt.tokenlogin(host, token, verify_ssl)
+    except Exception as ex:  # pylint: disable=broad-except
+        _LOGGER.error("Unable login to fgt Exception : %s", ex)
+
+    try:
+        scanner = FortiOSDeviceScanner(fgt)
+    except Exception as ex:  # pylint: disable=broad-except
+        _LOGGER.error("FortiOS get_scanner Initialize failed: %s", ex)
+        return False
+
+    return scanner
+
+
+class FortiOSDeviceScanner(DeviceScanner):
+    """This class queries a FortiOS unit for connected devices."""
+
+    def __init__(self, fgt) -> None:
+        """Initialize the scanner."""
+        self._clients = {}
+        self._clients_json = {}
+        self._fgt = fgt
+        self._update()
+
+    def _update(self):
+        """Update clients from the device."""
+        clients_json = (self._fgt.monitor('user/device/select',
+                                          ''))  # pylint: disable=W0105
```

Comment:
What does the emtpy string mean?

Reply:
This function external call is into fortiosapi. The REST API URL to FortiGate is constructed this way from fortiosapi.py:
`url_postfix = '/api/v2/monitor/' + path + '/' + name`
where 
path = 'user/device/select'
name = ''
Since 'user/device/select' does not require any 'name' it is empty.